### PR TITLE
Propagate RSUN_REF from Map into coordinate frame.

### PR DESCRIPTION
**Pull request by Cadair**
This is to ensure that you can correctly round-trip HPC-HGS-HPC, because if the scale on the image is based of a specific value of RSun then this needs to be used in the `calculate_distance` method of `HelioProjectiveFrame`

ping @mbobra @kbg

### CODE REVIEW

File: sunpy/coordinates/wcs_utils.py [hunk 1]
```diff
@@ -36,7 +41,7 @@ def solar_wcs_frame_mapping(wcs):
     ycoord = wcs.wcs.ctype[1][0:4]
 
     if xcoord == 'HPLN' and ycoord == 'HPLT':
-        return Helioprojective(dateobs=dateobs, observer=observer)
+        return Helioprojective(dateobs=dateobs, observer=observer, rsun=rsun)
```
Comment (dpshelio) [inline sunpy/coordinates/wcs_utils.py]: Is that all this needed?? without any change in frames to use it?? Wow!! that's amazing.

Comment (Cadair) [inline sunpy/coordinates/wcs_utils.py]: Yes, Helioprojective already had a rsun attribute it just wasn't getting auto filled by Map.

File: sunpy/coordinates/wcs_utils.py [hunk 2]
```diff
@@ -36,7 +41,7 @@ def solar_wcs_frame_mapping(wcs):
     ycoord = wcs.wcs.ctype[1][0:4]
 
     if xcoord == 'HPLN' and ycoord == 'HPLT':
```
Comment (wafels) [inline sunpy/coordinates/wcs_utils.py]: Why don't the other co-ordinate systems also carry observer and rsun?

Comment (Cadair) [inline sunpy/coordinates/wcs_utils.py]: That's more of a loaded question than it seems.

The short version is that they don't need them to represent the frame properly. Only HPC needs rsun and only HPC and HCC need observer location.

How `astropy.coordinates` works is that `SkyCoord` keeps track of these attributes (so if you go HPC->HG->HPC they will be set correctly, the idea of the frames is that they are self-contained and you only worry about round tripping and persistence of the attributes at the `SkyCoord` layer.

**Pull request merged**